app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -275,13 +281,22 @@ class WifiDirectManager @Inject constructor(
         fileReceiverDeviceAddress =
           if (isGroupOwner) selectedPeerDeviceInetAddress else groupOwnerAddress
         activity.toast(R.string.preparing_files, Toast.LENGTH_LONG)
-        senderDeviceAsyncTask = SenderDeviceAsyncTask(this, activity).also {
-          it.execute(*filesForTransfer.toTypedArray())
+        val senderDevice = SenderDevice(this, activity)
+        lifecycleCoroutineScope.launch {
+          val isFileSendSuccessfully = senderDevice.send(filesForTransfer)
+          if (isFileSendSuccessfully)
```

Comment:
when using an if use brackets unless it returns a value, also nesting 2 ifs is unnecessary when you can just && the conditions but also it is not accurate, you are only logging when it was successful and successful != complete which is what the log says. Really you can remove this if and just put this variable in the log statement

Reply:
done